### ISSUE TITLE
What is the expected request URL?

### ISSUE THREAD

**Issue by Jericho** [Pathoschild/FluentHttpClient#45]
Consider the following scenario:
```CSharp
var fluentClient = new FluentClient("http://api.vendor.com/v3/");
var request = client.Get("/myresource");
```

What is the expected request URL?

I'm asking because I expect the request to be sent to `http://api.vendor.com/v3/myresource` but instead the request is dispatched to `http://api.vendor.com/myresource`. The reason is because the resource starts with a slash. Of course, I could simply remove the slash and move on but I wanted to check what you think the request URL should be and whether this is an issue that must be addressed.

**Comment by Pathoschild** [comment]
The expected URL is `http://api.vendor.com/myresource`. The methods like `GetAsync` take a relative path rather than a segment, and relative paths starting with a slash are relative to the root. For example, `new FluentClient("http://api.vendor.com/v3/").GetAsync("../myresource")` is valid too.

**Comment by Jericho** [comment]
I see you point about `../myresource`. As far as a more "typical" scenario like `/myresource`, I was under the impression the base uri was indeed "the base" and would always be included in the final URL.

Oh well, I guess I need to change my expectation!!!

**Comment by Jericho** [comment]
Here's another scenario:
```CSharp
var fluentClient = new FluentClient("http://api.vendor.com/v3");
var request = client.Get("myresource");
```
I was very surprised that my request was dispatched to `http://api.vendor.com/myresource`. What happened to `v3`? I'm guessing this is due to the absence of a slash at the end of my base Uri and also absence of slash at the beginning of the resource???

These issues seem to be particularly evident when the base Uri is not the root of a web site. For example: your vendor URI contains version number.

It seems developers should be careful to follow these rules:
1. Make sure your base Uri ends with a slash
2. Make sure your resource Uri does not begin with a slash (unless you really want your request to be dispatched to the root and bypass the 'version' in the base Uri)

Do you think we should enforce these two rules in in order to help developers? For instance, we could enforce rule 1 in the fluent client constructor like so:
```CSharp
if (baseUri != null)
    this.BaseClient.BaseAddress = new Uri(baseUri.AbsoluteUri.TrimEnd('/') + '/');
```

**Comment by Pathoschild** [comment]
I'm not opposed, but we should be careful not to break valid requests in the process. For example, this code is valid but your fix would break it:

```cs
await new FluentClient("https://example.org/api.php").GetAsync("");
```

**Comment by Jericho** [comment]
Pseudo-code: 

If the last part of the baseUri contains a file extension, don't modify it
else make sure the uri ends with a slash

**Comment by Pathoschild** [comment]
Merged into `develop` for the upcoming 3.0 release. Thanks!
